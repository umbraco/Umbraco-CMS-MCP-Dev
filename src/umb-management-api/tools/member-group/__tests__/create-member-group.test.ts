import CreateMemberGroupTool from "../post/create-member-group.js";
import { MemberGroupTestHelper } from "./helpers/member-group-helper.js";
import { jest } from "@jest/globals";

const TEST_GROUP_NAME = "_Test Member Group Created";
const EXISTING_GROUP_NAME = "_Existing Member Group";

describe("create-member-group", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await MemberGroupTestHelper.cleanup(TEST_GROUP_NAME);
    await MemberGroupTestHelper.cleanup(EXISTING_GROUP_NAME);
  });

  it("should create a member group", async () => {
    const result = await CreateMemberGroupTool().handler({
      name: TEST_GROUP_NAME
    }, { signal: new AbortController().signal });

    expect(result).toMatchSnapshot();

    const items = await MemberGroupTestHelper.findMemberGroups(TEST_GROUP_NAME);
    expect(items).toEqual([{
      id: expect.any(String),
      name: TEST_GROUP_NAME
    }]);
  });

  it("should handle existing member group", async () => {
    // First create the group
    await CreateMemberGroupTool().handler({
      name: EXISTING_GROUP_NAME
    }, { signal: new AbortController().signal });

    // Try to create it again
    const result = await CreateMemberGroupTool().handler({
      name: EXISTING_GROUP_NAME
    }, { signal: new AbortController().signal });

    expect(result).toMatchSnapshot();
  });
}); 