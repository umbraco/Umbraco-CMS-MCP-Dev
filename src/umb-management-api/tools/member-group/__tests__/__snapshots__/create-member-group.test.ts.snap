// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-member-group should create a member group 1`] = `
{
  "content": [
    {
      "text": """",
      "type": "text",
    },
  ],
}
`;

exports[`create-member-group should handle existing member group 1`] = `
{
  "content": [
    {
      "text": "Error using create-member-group:
{
  "message": "Request failed with status code 400",
  "response": {
    "type": "Error",
    "title": "Duplicate name",
    "status": 400,
    "detail": "Another group with the same name already exists."
  }
}",
      "type": "text",
    },
  ],
}
`;
